### PR TITLE
BasicFaceRecognizer + EigenFaceRecognizer + FisherFaceRecognizer

### PR DESCRIPTION
**Pull request by diegohce**
Face recognizers collection with "some" smoke tests

### CODE REVIEW

File: contrib/face_recognizer.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 	"gocv.io/x/gocv"
 )
 
-type FaceRecognizer interface {
+type IFaceRecognizer interface {
```
Comment (deadprogram) [inline contrib/face_recognizer.go]: I think a better name for the interface would be just `FaceRecognizer`

File: contrib/face_recognizer.go [hunk 2]
```diff
@@ -24,6 +24,26 @@ type FaceRecognizer interface {
 	Close() error
 }
 
+type IBasicFaceRecognizer interface {
```
Comment (deadprogram) [inline contrib/face_recognizer.go]: I think a better name for the interface would be just `BasicFaceRecognizer`

File: contrib/face_recognizer.go (hunk 1 above)
Comment (deadprogram) [reply contrib/face_recognizer.go]: Or if that is just too hard with the `type FaceRecognizer` already, name it `FaceRecognize` and the other interface `BasicFaceRecognize`

File: contrib/face_recognizer.go (hunk 2 above)
Comment (deadprogram) [reply contrib/face_recognizer.go]: See above additional comment.

Comment (diegohce) [reply contrib/face_recognizer.go]: Yes! I forgot to change them back. Was trying to implement the interfaces more go-stylish.

**Comment by deadprogram** [comment]
Looks very promising @diegohce thank you!

What do you think about inserting the godocs links to the various OpenCV functions now?

**Comment by diegohce** [comment]
> Looks very promising @diegohce thank you!
> 
> What do you think about inserting the godocs links to the various OpenCV functions now?

Next PR will have the godocs links. sorry for that.

**Comment by deadprogram** [comment]
Thank you @diegohce now squashed/merged.

**Pull request merged**